// @railway/mcp-server/dist/cli/logs.test.js
import "../error-handling-mxVa5oOu.js";
import "../core-BMlc7cvF.js";
import "../projects-ENU6yIz7.js";
import "../version-BriONvFH.js";
import { buildLogCommand } from "../logs-Cvx68Kvp.js";
import { beforeEach, describe, globalExpect, it, vi } from "../vi.bdSIJ99Y-CjNqA1ej.js";

//#region src/cli/logs.test.ts
vi.mock("./version", () => ({ getCliFeatureSupport: vi.fn() }));
describe("Railway Logs Module", () => {
	beforeEach(() => {
		vi.clearAllMocks();
	});
	describe("buildLogCommand", () => {
		it("should build deployment command with lines and filter for v4.9.0+", async () => {
			const { getCliFeatureSupport } = await import("./version.js");
			vi.mocked(getCliFeatureSupport).mockResolvedValue({
				deployment: { list: true },
				logs: { args: {
					lines: true,
					filter: true
				} }
			});
			const command = await buildLogCommand({
				type: "deployment",
				lines: 100,
				filter: "error"
			});
			globalExpect(command).toBe("railway logs --deployment --lines 100 --filter \"error\"");
		});
		it("should build build command with lines and filter for v4.9.0+", async () => {
			const { getCliFeatureSupport } = await import("./version.js");
			vi.mocked(getCliFeatureSupport).mockResolvedValue({
				deployment: { list: true },
				logs: { args: {
					lines: true,
					filter: true
				} }
			});
			const command = await buildLogCommand({
				type: "build",
				lines: 200,
				filter: "warning"
			});
			globalExpect(command).toBe("railway logs --build --lines 200 --filter \"warning\"");
		});
		it("should default to 500 lines when json is false", async () => {
			const { getCliFeatureSupport } = await import("./version.js");
			vi.mocked(getCliFeatureSupport).mockResolvedValue({
				deployment: { list: true },
				logs: { args: {
					lines: true,
					filter: true
				} }
			});
			const command = await buildLogCommand({ type: "deployment" });
			globalExpect(command).toBe("railway logs --deployment --lines 500");
		});
		it("should default to 100 lines when json is true", async () => {
			const { getCliFeatureSupport } = await import("./version.js");
			vi.mocked(getCliFeatureSupport).mockResolvedValue({
				deployment: { list: true },
				logs: { args: {
					lines: true,
					filter: true
				} }
			});
			const command = await buildLogCommand({
				type: "deployment",
				json: true
			});
			globalExpect(command).toBe("railway logs --deployment --json --lines 100");
			globalExpect(command).toBe("railway logs --deployment --json --lines 100");
		});
		it("should not include lines/filter for older CLI versions", async () => {
			const { getCliFeatureSupport } = await import("./version.js");
			vi.mocked(getCliFeatureSupport).mockResolvedValue({
				deployment: { list: true },
				logs: { args: {
					lines: false,
					filter: false
				} }
			});
			const command = await buildLogCommand({
				type: "deployment",
				lines: 100,
				filter: "error"
			});
			globalExpect(command).toBe("railway logs --deployment");
		});
		it("should include deploymentId when provided", async () => {
			const { getCliFeatureSupport } = await import("./version.js");
			vi.mocked(getCliFeatureSupport).mockResolvedValue({
				deployment: { list: true },
				logs: { args: {
					lines: true,
					filter: true
				} }
			});
			const command = await buildLogCommand({
				type: "deployment",
				deploymentId: "deploy-123"
			});
			globalExpect(command).toBe("railway logs --deployment --lines 500 deploy-123");
		});
		it("should include service and environment when provided", async () => {
			const { getCliFeatureSupport } = await import("./version.js");
			vi.mocked(getCliFeatureSupport).mockResolvedValue({
				deployment: { list: true },
				logs: { args: {
					lines: true,
					filter: true
				} }
			});
			const command = await buildLogCommand({
				type: "deployment",
				service: "api",
				environment: "production",
				lines: 50
			});
			globalExpect(command).toBe("railway logs --deployment --lines 50 --service api --environment production");
		});
		it("should handle all parameters together", async () => {
			const { getCliFeatureSupport } = await import("./version.js");
			vi.mocked(getCliFeatureSupport).mockResolvedValue({
				deployment: { list: true },
				logs: { args: {
					lines: true,
					filter: true
				} }
			});
			const command = await buildLogCommand({
				type: "build",
				deploymentId: "deploy-456",
				service: "backend",
				environment: "staging",
				lines: 75,
				filter: "timeout"
			});
			globalExpect(command).toBe("railway logs --build --lines 75 --filter \"timeout\" deploy-456 --service backend --environment staging");
		});
		it("should not include --json flag when json is false", async () => {
			const { getCliFeatureSupport } = await import("./version.js");
			vi.mocked(getCliFeatureSupport).mockResolvedValue({
				deployment: { list: true },
				logs: { args: {
					lines: true,
					filter: true
				} }
			});
			const command = await buildLogCommand({
				type: "deployment",
				lines: 100,
				json: false
			});
			globalExpect(command).toBe("railway logs --deployment --lines 100");
		});
		it("should not include --json flag when undefined", async () => {
			const { getCliFeatureSupport } = await import("./version.js");
			vi.mocked(getCliFeatureSupport).mockResolvedValue({
				deployment: { list: true },
				logs: { args: {
					lines: true,
					filter: true
				} }
			});
			const command = await buildLogCommand({
				type: "deployment",
				lines: 100,
				json: false
			});
			globalExpect(command).toBe("railway logs --deployment --lines 100");
		});
		it("should include --json flag when true", async () => {
			const { getCliFeatureSupport } = await import("./version.js");
			vi.mocked(getCliFeatureSupport).mockResolvedValue({
				deployment: { list: true },
				logs: { args: {
					lines: true,
					filter: true
				} }
			});
			const command = await buildLogCommand({
				type: "deployment",
				lines: 100,
				json: true
			});
			globalExpect(command).toBe("railway logs --deployment --json --lines 100");
		});
	});
});

//#endregion
export {  };
