// @railway/mcp-server/dist/create-environment-DQmdFSZ2.js
import { createRailwayEnvironment } from "./environments-BMQ5EW9R.js";
import { createToolResponse } from "./utils-Cvxm88Gr.js";
import z from "zod";

//#region src/tools/create-environment.ts
const createEnvironmentTool = {
	name: "create-environment",
	title: "Create Environment",
	description: "Create a new Railway environment for the currently linked project. Optionally duplicate an existing environment and set service variables.",
	inputSchema: {
		workspacePath: z.string().describe("The path to the workspace where the environment should be created"),
		environmentName: z.string().describe("The name for the new environment"),
		duplicateEnvironment: z.string().optional().describe("The name of an existing environment to duplicate"),
		serviceVariables: z.array(z.object({
			service: z.string().describe("The service name or UUID"),
			variable: z.string().describe("The variable assignment (e.g., 'BACKEND_PORT=3000')")
		})).optional().describe("Service variables to assign in the new environment (only works when duplicating)")
	},
	handler: async ({ workspacePath, environmentName, duplicateEnvironment, serviceVariables }) => {
		try {
			const result = await createRailwayEnvironment({
				workspacePath,
				environmentName,
				duplicateEnvironment,
				serviceVariables
			});
			return createToolResponse(result);
		} catch (error) {
			const errorMessage = error instanceof Error ? error.message : "Unknown error occurred";
			return createToolResponse(`❌ Failed to create environment

**Error:** ${errorMessage}\n\n**Next Steps:**
• Ensure you have a Railway project linked
• Check that the environment name is valid and unique
• Verify you have permissions to create environments in this project
• If duplicating, ensure the source environment exists
• If using service variables, ensure the service exists in the source environment`);
		}
	}
};

//#endregion
export { createEnvironmentTool };
