// @railway/mcp-server/dist/deploy-template-CE3DJG5u.js
import { getRailwayAuthToken } from "./auth-fMu9JRb4.js";
import Fuse from "fuse.js";
import { GraphQLClient } from "graphql-request";

//#region src/api/deploy-template.ts
const searchAndListTemplates = async ({ searchQuery }) => {
	const query = `
		query {
			templates {
				edges {
					node {
						id
						name
						description
						category
						serializedConfig
						activeProjects
						health
						totalPayout
						isVerified
					}
				}
			}
		}
	`;
	try {
		const templates = (await new GraphQLClient("https://backboard.railway.com/graphql/v2", { headers: { "x-source": "railway-mcp-server" } }).request(query)).templates.edges.map((edge) => edge.node);
		const totalCount = templates.length;
		const sortedTemplates = templates.sort((a, b) => {
			if (a.isVerified && !b.isVerified) return -1;
			if (!a.isVerified && b.isVerified) return 1;
			if (a.totalPayout !== b.totalPayout) return (b.totalPayout || 0) - (a.totalPayout || 0);
			if (a.activeProjects !== b.activeProjects) return (b.activeProjects || 0) - (a.activeProjects || 0);
			if (a.health !== b.health) return (b.health || 0) - (a.health || 0);
			return 0;
		});
		if (!searchQuery || searchQuery.trim() === "") return {
			templates: sortedTemplates,
			filteredCount: totalCount,
			totalCount
		};
		const filteredTemplates = new Fuse(sortedTemplates, {
			keys: [
				"name",
				"description",
				"category"
			],
			threshold: .3,
			includeScore: true,
			includeMatches: true
		}).search(searchQuery.trim()).map((result) => result.item);
		return {
			templates: filteredTemplates,
			filteredCount: filteredTemplates.length,
			totalCount
		};
	} catch (error) {
		if (error instanceof Error) throw new Error(`Failed to search Railway templates: ${error.message}`);
		throw new Error("Failed to search Railway templates: Unknown error");
	}
};
const deployTemplate = async ({ environmentId, projectId, serializedConfig, templateId, teamId }) => {
	const query = `
		mutation deployTemplate($environmentId: String, $projectId: String, $templateId: String!, $teamId: String, $serializedConfig: SerializedTemplateConfig!) {
			templateDeployV2(input: {
				environmentId: $environmentId,
				projectId: $projectId,
				templateId: $templateId,
				teamId: $teamId,
				serializedConfig: $serializedConfig
			}) {
				projectId
				workflowId
			}
		}
	`;
	try {
		const token = getRailwayAuthToken();
		return (await new GraphQLClient("https://backboard.railway.com/graphql/v2", { headers: {
			Authorization: `Bearer ${token}`,
			"x-source": "railway-mcp-server"
		} }).request(query, {
			environmentId,
			projectId,
			templateId,
			teamId,
			serializedConfig
		})).templateDeployV2;
	} catch (error) {
		if (error instanceof Error) throw new Error(`Failed to deploy template: ${error.message}`);
		throw new Error("Failed to deploy template: Unknown error");
	}
};

//#endregion
export { deployTemplate, searchAndListTemplates };
