// @railway/mcp-server/dist/generate-domain-CZNdFeJ5.js
import { generateRailwayDomain } from "./domain-CMz5XAyW.js";
import { createToolResponse } from "./utils-Cvxm88Gr.js";
import z from "zod";

//#region src/tools/generate-domain.ts
const generateDomainTool = {
	name: "generate-domain",
	title: "Generate Railway Domain",
	description: "Generate a domain for the currently linked Railway project. If a domain already exists, it will return the existing domain URL. Optionally specify a service to generate the domain for.",
	inputSchema: {
		workspacePath: z.string().describe("The path to the workspace to generate domain for"),
		service: z.string().optional().describe("The name of the service to generate the domain for (optional)")
	},
	handler: async ({ workspacePath, service }) => {
		try {
			const domain = await generateRailwayDomain({
				workspacePath,
				service
			});
			return createToolResponse(`✅ Successfully generated Railway domain${service ? ` for service '${service}'` : ""}:\n\n🚀 ${domain}\n\n**Note:** This domain is now available for your Railway project.`);
		} catch (error) {
			const errorMessage = error instanceof Error ? error.message : "Unknown error occurred";
			return createToolResponse(`❌ Failed to generate Railway domain

**Error:** ${errorMessage}\n\n**Next Steps:**
• Ensure you have a Railway project linked
• Check that you have permissions to generate domains
• Verify the project has been deployed at least once
• Run \`railway link\` to ensure proper project connection`);
		}
	}
};

//#endregion
export { generateDomainTool };
