// @railway/mcp-server/dist/link-service-ChMe6sIU.js
import { getRailwayServices, linkRailwayService } from "./services-BXZqOgoN.js";
import { createToolResponse } from "./utils-Cvxm88Gr.js";
import z from "zod";

//#region src/tools/link-service.ts
const linkServiceTool = {
	name: "link-service",
	title: "Link Railway Service",
	description: "Link a service to the current Railway project. If no service is specified, it will list available services",
	inputSchema: {
		workspacePath: z.string().describe("The path to the workspace to link the service to"),
		serviceName: z.string().optional().describe("The service name to link")
	},
	handler: async ({ workspacePath, serviceName }) => {
		try {
			if (serviceName) {
				const result = await linkRailwayService({
					workspacePath,
					serviceName
				});
				return createToolResponse(`✅ Successfully linked service '${serviceName}':\n\n${result}`);
			} else {
				const servicesResult = await getRailwayServices({ workspacePath });
				if (!servicesResult.success) return createToolResponse(`❌ Failed to get Railway services

**Error:** ${servicesResult.error}\n\n**Next Steps:**
• Ensure you have a Railway project linked
• Check that you have permissions to view services
• Run \`railway link\` to ensure proper project connection`);
				if (!servicesResult.services || servicesResult.services.length === 0) return createToolResponse("ℹ️ No services found in this project. Create a service first.");
				const result = `Available services:\n${servicesResult.services.map((s) => `- ${s}`).join("\n")}\n\nRun with a service name to link it.`;
				return createToolResponse(result);
			}
		} catch (error) {
			const errorMessage = error instanceof Error ? error.message : "Unknown error occurred";
			return createToolResponse(`❌ Failed to link Railway service

**Error:** ${errorMessage}\n\n**Next Steps:**
• Ensure you have a Railway project linked
• Check that the service name is correct
• Verify you have permissions to link services
• Run \`railway link\` to ensure proper project connection`);
		}
	}
};

//#endregion
export { linkServiceTool };
