// @railway/mcp-server/dist/link-environment-DyCObkhO.js
import { linkRailwayEnvironment } from "./environments-BMQ5EW9R.js";
import { createToolResponse } from "./utils-Cvxm88Gr.js";
import z from "zod";

//#region src/tools/link-environment.ts
const linkEnvironmentTool = {
	name: "link-environment",
	title: "Link Environment",
	description: "Link to a specific Railway environment. If no environment is specified, it will list available environments for selection.",
	inputSchema: {
		workspacePath: z.string().describe("The path to the workspace to link the environment to"),
		environmentName: z.string().describe("The environment name to link to")
	},
	handler: async ({ workspacePath, environmentName }) => {
		try {
			const result = await linkRailwayEnvironment({
				workspacePath,
				environmentName
			});
			return createToolResponse(result);
		} catch (error) {
			const errorMessage = error instanceof Error ? error.message : "Unknown error occurred";
			return createToolResponse(`❌ Failed to link environment

**Error:** ${errorMessage}\n\n**Next Steps:**
• Ensure you have a Railway project linked
• Check that the environment name is correct
• Run \`railway environment\` to see available environments`);
		}
	}
};

//#endregion
export { linkEnvironmentTool };
