// @railway/mcp-server/dist/list-variables-CFphrMwZ.js
import { listRailwayVariables } from "./variables-BKzFECGf.js";
import { createToolResponse } from "./utils-Cvxm88Gr.js";
import z from "zod";

//#region src/tools/list-variables.ts
const listVariablesTool = {
	name: "list-variables",
	title: "List Railway Variables",
	description: "Show variables for the active environment",
	inputSchema: {
		workspacePath: z.string().describe("The path to the workspace to list variables from"),
		service: z.string().optional().describe("The service to show variables for (optional)"),
		environment: z.string().optional().describe("The environment to show variables for (optional)"),
		kv: z.boolean().optional().describe("Show variables in KV format (optional)"),
		json: z.boolean().optional().describe("Output in JSON format (optional)")
	},
	handler: async ({ workspacePath, service, environment, kv, json }) => {
		try {
			const variables = await listRailwayVariables({
				workspacePath,
				service,
				environment,
				kv,
				json
			});
			return createToolResponse(variables);
		} catch (error) {
			const errorMessage = error instanceof Error ? error.message : "Unknown error occurred";
			return createToolResponse(`❌ Failed to list Railway variables

**Error:** ${errorMessage}\n\n**Next Steps:**
• Ensure you have a Railway project linked
• Check that the service and environment exist
• Verify you have permissions to view variables
• Run \`railway link\` to ensure proper project connection`);
		}
	}
};

//#endregion
export { listVariablesTool };
