// @railway/mcp-server/dist/list-deployments-DGamMLAf.js
import { listDeployments } from "./deployment-DXcNTO8k.js";
import { createToolResponse } from "./utils-Cvxm88Gr.js";
import z from "zod";

//#region src/tools/list-deployments.ts
const listDeploymentsTool = {
	name: "list-deployments",
	title: "List Railway Deployments",
	description: "List deployments for a Railway service with IDs, statuses and other metadata. Requires Railway CLI v4.10.0+.",
	inputSchema: {
		workspacePath: z.string().describe("The path to the workspace to list deployments from"),
		service: z.string().optional().describe("Service name or ID to list deployments for (defaults to linked service)"),
		environment: z.string().optional().describe("Environment to list deployments from (defaults to linked environment)"),
		limit: z.number().min(1).max(1e3).default(20).optional().describe("Maximum number of deployments to show (default: 20, max: 1000)"),
		json: z.boolean().default(false).optional().describe("Return deployments as structured JSON data. When true, the output will contain ids, statuses, and other metadata")
	},
	handler: async (options) => {
		try {
			const result = await listDeployments(options);
			if (!result.success) throw new Error(result.error);
			return createToolResponse(result.output);
		} catch (error) {
			const errorMessage = error instanceof Error ? error.message : "Unknown error occurred";
			return createToolResponse(`❌ Failed to list Railway deployments

**Error:** ${errorMessage}\n\n**Next Steps:**
• Ensure you are logged into Railway CLI (\`railway login\`)
• Check that you have a project linked (\`railway link\`)
• Verify you have permissions to view deployments
• Try running \`railway login\` to refresh your authentication`);
		}
	}
};

//#endregion
export { listDeploymentsTool };
