// @railway/mcp-server/dist/set-variables-CUqHEIwy.js
import { setRailwayVariables } from "./variables-BKzFECGf.js";
import { createToolResponse } from "./utils-Cvxm88Gr.js";
import z from "zod";

//#region src/tools/set-variables.ts
const setVariablesTool = {
	name: "set-variables",
	title: "Set Railway Variables",
	description: "Set environment variables for the active environment",
	inputSchema: {
		workspacePath: z.string().describe("The path to the workspace to set variables in"),
		variables: z.array(z.string()).describe("Array of '{key}={value}' environment variable pairs to set"),
		service: z.string().optional().describe("The service to set variables for (optional)"),
		environment: z.string().optional().describe("The environment to set variables for (optional)"),
		skipDeploys: z.boolean().optional().describe("Skip triggering deploys when setting variables (optional)")
	},
	handler: async ({ workspacePath, variables, service, environment, skipDeploys }) => {
		try {
			const result = await setRailwayVariables({
				workspacePath,
				variables,
				service,
				environment,
				skipDeploys
			});
			return createToolResponse(result);
		} catch (error) {
			const errorMessage = error instanceof Error ? error.message : "Unknown error occurred";
			return createToolResponse(`❌ Failed to set Railway variables

**Error:** ${errorMessage}\n\n**Next Steps:**
• Ensure you have a Railway project linked
• Check that the service and environment exist
• Verify you have permissions to set variables
• Ensure variable format is correct (KEY=value)
• Run \`railway link\` to ensure proper project connection`);
		}
	}
};

//#endregion
export { setVariablesTool };
