// @railway/mcp-server/dist/version-BriONvFH.js
import { runRailwayCommand } from "./core-BMlc7cvF.js";
import * as semver from "semver";

//#region src/cli/version.ts
let cachedVersion = null;
let versionCacheTime = null;
const VERSION_CACHE_DURATION = 300 * 1e3;
/**
* Get the Railway CLI version
* @returns Version string (e.g., "4.9.0") or null if unable to determine
*/
const getRailwayVersion = async () => {
	if (cachedVersion && versionCacheTime && Date.now() - versionCacheTime < VERSION_CACHE_DURATION) return cachedVersion;
	try {
		const { stdout } = await runRailwayCommand("railway --version");
		const version = semver.coerce(stdout);
		if (version) {
			cachedVersion = version.version;
			versionCacheTime = Date.now();
			return cachedVersion;
		}
	} catch (_error) {}
	return null;
};
/** Clear the cached version  */
const clearVersionCache = () => {
	cachedVersion = null;
	versionCacheTime = null;
};
/**Force a fresh version check */
const refreshRailwayVersion = async () => {
	clearVersionCache();
	return getRailwayVersion();
};
/**
* Compare version strings using semver
* @returns 1 if version1 > version2, -1 if version1 < version2, 0 if equal
*/
const compareVersions = (version1, version2) => {
	return semver.compare(version1, version2);
};
/** Get Railway CLI feature support based on version */
const getCliFeatureSupport = async () => {
	const version = await getRailwayVersion();
	const supportsLogFeatures = version ? semver.gte(version, "4.9.0") : false;
	const supportsDeploymentList = version ? semver.gte(version, "4.10.0") : false;
	return {
		logs: { args: {
			lines: supportsLogFeatures,
			filter: supportsLogFeatures
		} },
		deployment: { list: supportsDeploymentList }
	};
};

//#endregion
export { clearVersionCache, compareVersions, getCliFeatureSupport, getRailwayVersion, refreshRailwayVersion };
